src/generic/Matrix.jl

```diff
@@ -490,7 +490,7 @@ Base.IteratorEltype(::Type{<:MatrixElem}) = Base.HasEltype() # default
 #
 ###############################################################################
 
-function setindex!(a::AbstractAlgebra.MatElem{T}, b::AbstractAlgebra.MatElem{T}, r::UnitRange{Int}, c::UnitRange{Int}) where T
+function setindex!(a::MatrixElem{T}, b::MatrixElem{<:RingElement}, r::UnitRange{Int}, c::UnitRange{Int}) where T
```

Comment:
Out of curiosity, is the `<:RingElement` necessary here for dispatch or something?